### PR TITLE
Omit bookmarked files

### LINKED ISSUE #67: [Enhancement] Omit Bookmarked Files
Is it possible to add a way to omit all the bookmarked files from the recent files list?
I always have the bookmarks list open, so having them show up in the recent files is redundant.

### PR DESCRIPTION
**Pull request by zayKenyon**
I appreciate the helpfulness in #87 , I've given #67 a go. 

Couldn't find documentation for Bookmarks, as always still learning, happy to make adjustments.

Fixes #67 .

### CODE REVIEW

File: main.ts [hunk 1]
```diff
@@ -22,10 +22,17 @@ interface FilePath {
...
+  ctime: number;
+  path: string;
+  type: string;
+}
+
 interface RecentFilesData {
   recentFiles: FilePath[];
   omittedPaths: string[];
   omittedTags: string[];
+  omitBookmarks: boolean
```
Comment (zayKenyon) [inline main.ts:35]: Whoops, forgot the semi-colon, I'll fix it in the PR-suggestions commit.

File: main.ts [hunk 2]
```diff
@@ -395,6 +403,16 @@ export default class RecentFilesPlugin extends Plugin {
       }
     }
 
+    // Matches for Bookmarks
+    if (tfile && this.data.omitBookmarks) {
+      const bookmarkedFiles: [BookmarkedFile] = this.app.internalPlugins.plugins.bookmarks.instance.items;
```
Comment (zayKenyon) [inline main.ts:408]: What is the correct way to get bookmarks? I was hesitant to commit a ts-ignore.

**Comment by zayKenyon** [comment]
When I tested my suggested way of retrieving the bookmarked items with Bookmarks disabled it returned an empty list, however I prefer your method because it allows files that were bookmarked when the plugin was enabled to be added to the Recent Files list once the Bookmarks plugin is disabled. 

Long winded way to say: 👍

File: main.ts [hunk 3]
```diff
@@ -404,8 +404,10 @@ export default class RecentFilesPlugin extends Plugin {
     }
 
     // Matches for Bookmarks
-    if (tfile && this.data.omitBookmarks) {
-      const bookmarkedFiles: [BookmarkedFile] = this.app.internalPlugins.plugins.bookmarks.instance.items;
+    // @ts-ignore
+    const bookmarksPlugin = this.app.internalPlugins.getEnabledPluginById('bookmarks')
+    if (tfile && this.data.omitBookmarks && bookmarksPlugin) {
+      const bookmarkedFiles: BookmarkedFile[] = bookmarksPlugin.items;
       const bookmarkedPaths = bookmarkedFiles.map(({ path }) => ( path ));
```
Comment (tgrosinger) [inline main.ts:411]: One more improvement I just noticed. Right now we're iterating the list twice. Once to extract the path, and then again to see if our path is in it. We could combine those to make it a bit more efficient.

```ts
const bookmarkedFiles: BookmarkedFile[] = bookmarksPlugin.items;
if (bookmarkedFiles.some(({ path }) => path === tfile.path)) {
    return false;
}
```

Comment (zayKenyon) [reply main.ts:411]: Clever!

**Review by tgrosinger — APPROVED** [review]

**Comment by tgrosinger** [comment]
Thank you!

**Pull request merged**